### ISSUE TITLE
Getting 999 when hitting public profile pages in linkedIn.

### ISSUE THREAD

**Issue by Karan-Daiict** [yatish27/linkedin-scraper#105]
When I hit public profiles I get 999. I guess after 4 to 5 continous hits, linkedIn blocks the ip address. So changed the ip, but still 999 exists. Is it possible that they are able to get local machine ip address?
How to solve this error without login?

I'm stuck with this. Please help.

**Comment by rochenka** [comment]
@Karan-Daiict  @adolfopeccin same for me until I found an API that avoids the 999, i make GET requests to that API which calls Linkedin and returns the data, the API is from proxycrawl.

**Comment by rochenka** [comment]
@Karan-Daiict I do not use Linkedin API, i use ProxyCrawl API which calls linkedin internally. https://proxycrawl.com/scraping-api-avoid-captchas-blocks

**Comment by Karan-Daiict** [comment]
Ohh okay, that is good. Scraping with proxy built in. But you'll have to
pay for it.

**Comment by rochenka** [comment]
@Karan-Daiict yes I do not want to manage headless browser infrastructure or whatever tricks comes along the way from Linkedin, especially that i am building a project for recruiting so I need data constantly, that is why I pay for it, the other option is to find proxies that work for Linkedin which is another difficult task to do and also I have to pay for it, so I went for an easy good solution. Do you use proxies or how do you get the data?

**Comment by Karan-Daiict** [comment]
Yeah. Basically I created some proxies and ran them in round robin fashion.
Does the API give complete data and is the signature easy to use for proxy
crawl? Also I think LinkedIn will block based on creds. Do check that.

**Comment by rochenka** [comment]
I do not really know what they do internally, i just receive the html data that is public, i do not think they support private data crawling from linkedin though.

**Comment by adolfopeccin** [comment]
@rochenka how did you implemented the scraping for linkedin data using proxycrawl?

**Comment by rochenka** [comment]
@adolfopeccin Its basically simple, proxycrawl does the crawling with their API, you need to have a token. I think you can get one token for 1000 free requests. So, proxycrwal  gives me the raw html data response and I then use linkedin-scraper to scrape the html content.

**Comment by abhishmitra** [comment]
@rochenka 

Hi,
I spoke to proxycrawl but they have certain limits and stuff. Did they impose those recently? How are they working for you?

**Comment by Bilal815** [comment]
I recently tried Crawlbase which was ProxyCrawl previously to scrape LinkedIn and it worked. But since I read some of their docs, it says that you need JS enabled to scrape/crawl dynamic sites and that's what's paid. This is where it gets interesting https://crawlbase.com/docs/crawling-api/scrapers/#linkedin. Just make sure to signup.

**Comment by Bilal815** [comment]
> @rochenka
> 
> Hi, I spoke to proxycrawl but they have certain limits and stuff. Did they impose those recently? How are they working for you?


I recently tried Crawlbase which was ProxyCrawl previously to scrape LinkedIn and it worked. But since I read some of their docs, it says that you need JS enabled to scrape/crawl dynamic sites and that's what's paid. This is where it gets interesting https://crawlbase.com/docs/crawling-api/scrapers/#linkedin. Just make sure to signup.

**Comment by Bilal815** [comment]
> @rochenka
> 
> Hi, I spoke to proxycrawl but they have certain limits and stuff. Did they impose those recently? How are they working for you?

I recently tried Crawlbase which was ProxyCrawl previously and it worked like a charm! Looks like these guys know what they are going for. They say that with JS-enabled for dynamic content which is a need for LinkedIn, you need to pay them a little. I tried my luck and it got interesting here https://crawlbase.com/docs/crawling-api/scrapers/#linkedin.

**Comment by Bilal815** [comment]
> @rochenka
> 
> Hi, I spoke to proxycrawl but they have certain limits and stuff. Did they impose those recently? How are they working for you?

I recently tried Crawlbase which was ProxyCrawl previously and it worked like a charm! Looks like these guys know what they are going for. They say that with JS-enabled for dynamic content which is a need for LinkedIn, you need to pay them a little. I tried my luck and it got interesting here https://crawlbase.com/docs/crawling-api/scrapers/#linkedin.

**Comment by Bilal815** [comment]
> @rochenka
> 
> Hi, I spoke to proxycrawl but they have certain limits and stuff. Did they impose those recently? How are they working for you?

I recently tried Crawlbase which was ProxyCrawl previously. They say that with JS-enabled for dynamic content which is a need for LinkedIn, you need to pay them a little. I tried my luck and it got interesting here https://crawlbase.com/docs/crawling-api/scrapers/#linkedin.
